src/main/java/io/github/moulberry/notenoughupdates/listener/RenderListener.java

```diff
@@ -748,6 +758,26 @@ private void renderDungeonChestOverlay(GuiScreen gui) {
 					for (int i = 0; i < 5; i++) {
 						ItemStack item = lower.getStackInSlot(11 + i);
 						String internal = neu.manager.getInternalNameForItem(item);
+						String displayName = item.getDisplayName();
+						if (displayName.contains(" Essence §8x") && neu.config.dungeons.useEssenceCostFromBazaar) {
+							String type = io.github.moulberry.notenoughupdates.core.util.StringUtils.substringBetween(
```

Comment:
It would require importing StringUtils but
```suggestion
							String type = StringUtils.substringBetween(
```

Reply:
Doing this whole check with regex now